engine/vcs/types.go

```diff
@@ -84,9 +84,10 @@ var errGithubConfigurationError = fmt.Errorf("Github configuration Error")
 
 // GitlabServerConfiguration represents the gitlab configuration
 type GitlabServerConfiguration struct {
-	AppID  string `toml:"appId" json:"-" comment:"#######\n CDS <-> Gitlab. Documentation on https://ovh.github.io/cds/hosting/repositories-manager/gitlab/ \n#######"`
-	Secret string `toml:"secret" json:"-"`
-	Status struct {
+	AppID            string `toml:"appId" json:"-" comment:"#######\n CDS <-> Gitlab. Documentation on https://ovh.github.io/cds/hosting/repositories-manager/gitlab/ \n#######"`
+	Secret           string `toml:"secret" json:"-"`
+	OAuthCallbackURL string `toml:"callbackUrl" json:"OAuth Application Callback URL"`
```

Comment:
Can you add a default value with http://localhost:8081/repositories_manager/oauth2/callback, to generate default config that works for dev purpose.

Reply:
json:"OAuth Application Callback URL" --> this is probably the "comment" annotation, not the json annotation :)

Reply:
fixed